src/Sarif.Viewer.VisualStudio/ErrorList/SarifAbsentResultsTableDataSource.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.TableManager;
+
+namespace Microsoft.Sarif.Viewer.ErrorList
+{
+    // A "stub" data source whose sole purpose is to cause the Category column
+    // to be displayed. It never actually adds entries to the Error List.
+    //
+    // For an explanation of why these this source is necessary, see the comment
+    // near the top of SarifResultTableEntry.cs
+    internal class SarifAbsentResultsTableDataSource : SarifTableDataSourceBase
+    {
+        private static SarifAbsentResultsTableDataSource _instance;
+
+        private SarifAbsentResultsTableDataSource()
+        {
+            if (!SarifViewerPackage.IsUnitTesting)
+            {
+#pragma warning disable VSTHRD108 // Assert thread affinity unconditionally
+                ThreadHelper.ThrowIfNotOnUIThread();
+#pragma warning restore VSTHRD108 // Assert thread affinity unconditionally
```

Comment:
>// Assert thread affinity unconditionally [](start = 34, length = 41)

More style, I personally don't find this duplication of rule description provides for better readability when scope is limited to a single line like this. I can appreciate that this is the default code emit behavior, but I usually remove these. My care factor is very small on this point, though.

Reply:
Changed everywhere.

---
In reply to: [538750192](https://github.com/microsoft/sarif-visualstudio-extension/pull/313#discussion_r538750192) [](ancestors = 538750192)